uxsdcxx/capnp.py

```diff
@@ -207,16 +207,32 @@ def render_capnp_file(schema: UxsdSchema, cmdline: str, input_file: str) -> str:
 def load_fn_from_element(e: UxsdElement) -> str:
 	out = ""
 	out += "template <class T, typename Context>\n"
-	out += "inline void load_%s_capnp(T &out, kj::ArrayPtr<const ::capnp::word> data, Context &context){\n" % e.name
+	out += "inline void load_%s_capnp(T &out, kj::ArrayPtr<const ::capnp::word> data, Context &context, const char * filename){\n" % e.name
 
 	out += "\t/* Increase reader limit to 1G words. */\n"
 	out += "\t::capnp::ReaderOptions opts = ::capnp::ReaderOptions();\n"
 	out += "\topts.traversalLimitInWords = 1024 * 1024 * 1024;\n"
 	out += "\t::capnp::FlatArrayMessageReader reader(data, opts);\n"
 	out += "\tauto root = reader.getRoot<ucap::{}>();\n".format(to_type(e))
+	out += "\tstd::vector<std::pair<const char*, size_t>> stack;\n"
+	out += "\tstack.reserve(20);\n"
```

Comment:
It might be good to give the 20 a name: `ERROR_STACK_DEPTH`?

Reply:
at least in the python code

Reply:
Done.  In theory the code could correctly determine maximum depth, but I'm not stressed about this being wrong, and stack depth of 20 is pretty conservative.